backend_new/src/services/user.service.ts

```diff
@@ -0,0 +1,234 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { User } from '../dtos/users/user.dto';
+import { mapTo } from '../utilities/mapTo';
+import {
+  buildPaginationMetaInfo,
+  calculateSkip,
+  calculateTake,
+} from '../utilities/pagination-helpers';
+import { buildOrderBy } from '../utilities/build-order-by';
+import { Prisma } from '@prisma/client';
+import { UserQueryParams } from '../dtos/users/user-query-param.dto';
+import { PaginatedUserDto } from '../dtos/users/paginated-user.dto';
+import { OrderByEnum } from '../enums/shared/order-by-enum';
+import { UserFilterKeys } from '../enums/user_accounts/filter-key-enum';
+
+/*
+  this is the service for users
+  it handles all the backend's business logic for reading/writing/deleting user data
+*/
+
+const view: Prisma.UserAccountsInclude = {
+  listings: true,
+  jurisdictions: true,
+  userRoles: true,
+};
+
+@Injectable()
+export class UserService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of users given the params passed in
+  */
+  async list(params: UserQueryParams, user: User): Promise<PaginatedUserDto> {
+    const whereClause = this.buildWhereClause(params, user);
+    const count = await this.prisma.userAccounts.count({
+      where: whereClause,
+    });
+
+    const rawUsers = await this.prisma.userAccounts.findMany({
+      skip: calculateSkip(params.limit, params.page),
+      take: calculateTake(params.limit),
+      orderBy: buildOrderBy(
+        ['firstName', 'lastName'],
+        [OrderByEnum.ASC, OrderByEnum.ASC],
+      ),
+      include: view,
+      where: whereClause,
+    });
```

Comment:
@YazeedLoonat , I was curious if we could reduce the findMany and count to a single transaction, and it looks like there is a way: https://stackoverflow.com/questions/74328100/prisma-find-many-and-count-in-one-request. Can you please test this out and make sure Prisma handles this well.

Reply:
So after doing some digging I found some interesting comments:

https://github.com/prisma/prisma/issues/7550#issuecomment-1594572700
and  https://www.prisma.io/blog/how-prisma-supports-transactions-x45s1d5l0ww1

from my reading of it $transaction still is happening sequentially, I'll switch to a promise.all style solution since the results of 1 query doesn't affect the 2nd query at all

It seems like eventually prisma will have a built in solution for this but for the time being we can run with the promise.all approach for stuff like this or transactions in other cases!